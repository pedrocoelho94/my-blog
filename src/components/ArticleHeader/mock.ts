import { data } from 'api/mockPosts.json'

const { id, title, excerpt, cover, author, categories, createdAt } =
  data.posts[0]

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}
