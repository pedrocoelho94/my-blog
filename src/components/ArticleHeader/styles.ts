import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
