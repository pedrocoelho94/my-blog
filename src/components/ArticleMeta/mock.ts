export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'pedro-coelho',
    displayName: 'Pedro Coelho'
  },
  categories: [
    {
      id: '78724ec23f',
      displayName: 'Tech',
      slug: 'tech'
    },
    {
      id: '762vg214g34',
      displayName: 'JS',
      slug: 'javascript'
    }
  ]
}
