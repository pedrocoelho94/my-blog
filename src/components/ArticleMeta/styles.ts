import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
