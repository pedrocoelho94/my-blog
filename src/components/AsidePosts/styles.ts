import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
