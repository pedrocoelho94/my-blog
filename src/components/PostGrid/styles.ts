import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;
  margin: 0 auto;
`

export const NotFound = styled.p`
  ${({ theme }) => css`
    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.medium};
  `}
`

export const Grid = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    display: grid;
    grid-template-columns: 1fr;
    gap: ${theme.spacings.xhuge};
  `}
`
