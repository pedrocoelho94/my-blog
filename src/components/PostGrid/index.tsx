import PostCard, { PostCardProps } from 'components/PostCard'
import * as S from './styles'

export type PostGridProps = {
  posts: PostCardProps[]
  twoColumns?: boolean
}

const PostGrid = ({ posts = [], twoColumns = false }: PostGridProps) => (
  <S.Wrapper>
    {posts.length === 0 && <S.NotFound>Nenhum post encontrado aqui</S.NotFound>}

    <S.Grid twoColumns={twoColumns}>
      {posts.length > 0 &&
        posts.map((post) => <PostCard key={`post-${post.id}`} {...post} />)}
    </S.Grid>
  </S.Wrapper>
)

export default PostGrid
