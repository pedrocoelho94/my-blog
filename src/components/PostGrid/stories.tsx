import { Story, Meta } from '@storybook/react'
import PostGrid, { PostGridProps } from '.'

import mock from './mock'

export default {
  title: 'PostGrid',
  component: PostGrid,
  args: mock
} as Meta

export const Default: Story<PostGridProps> = (args) => (
  <div style={{ maxWidth: '98rem', margin: '0 auto' }}>
    <PostGrid {...args} />
  </div>
)

export const WithoutPosts: Story<PostGridProps> = (args) => (
  <div style={{ maxWidth: '98rem', margin: '0 auto' }}>
    <PostGrid {...args} />
  </div>
)

WithoutPosts.args = {
  posts: []
}
