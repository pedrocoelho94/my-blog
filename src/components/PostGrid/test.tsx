import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'

import PostGrid, { PostGridProps } from '.'
import mock from './mock'

const props: PostGridProps = mock

describe('<PostGrid />', () => {
  it('should render the heading', () => {
    renderWithTheme(<PostGrid {...props} posts={[]} />)

    expect(screen.getByText(/Nenhum post/i)).toBeInTheDocument()
  })

  it('should render two posts', () => {
    const { container } = renderWithTheme(<PostGrid {...props} />)

    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument()

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getAllByRole('img')).toHaveLength(2)
    expect(container.querySelectorAll('p')).toHaveLength(2)

    expect(container).toMatchSnapshot()
  })
})
