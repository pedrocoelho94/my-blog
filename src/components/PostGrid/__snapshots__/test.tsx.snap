// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should render two posts 1`] = `
.c8 {
  color: #000000;
  line-height: 1.2;
  font-size: calc(2.4rem * 0.75);
  text-transform: none;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: grid;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-template-columns: 1fr 1fr;
  gap: 2.4rem;
}

.c2 .c7 {
  color: #333333;
  margin-top: 0;
  margin-bottom: 1.6rem;
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c2 a:hover {
  color: #dc143c;
}

.c3 {
  position: relative;
  padding-bottom: 56.25%;
  max-width: 100%;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
  overflow: hidden;
}

.c4:hover {
  opacity: 0.8;
}

.c5 {
  opacity: 0.6;
  margin-bottom: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c5 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 span > svg {
  margin-right: 0.8rem;
}

.c9 {
  max-width: 100%;
  font-size: calc(0.8rem * 1.9);
}

.c0 {
  width: 100%;
  margin: 0 auto;
}

.c1 {
  margin: 0 auto;
  display: grid;
  grid-template-columns: 1fr;
  gap: 6.4rem;
}

@media (max-width:768px) {
  .c8 {
    font-size: calc(1.6rem * 1.2);
  }
}

@media (max-width:600px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:1170px) {
  .c5 {
    font-size: calc(0.8rem * 1.5);
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <a
            href="/futebol/al-sadd-anuncia-acordo-para-liberar-xavi-mas-barcelona-nega"
          >
            <img
              alt=""
              class="c4"
              src="https://res.cloudinary.com/rabbitz/image/upload/v1636129235/Xavi_Hernandez_Barcelona_c60f5b3e16.jpg"
            />
          </a>
        </div>
        <div
          class=""
        >
          <div
            class="c5"
          >
            <span>
              <svg
                aria-hidden="true"
                class="c6"
                fill="currentColor"
                focusable="false"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
              <a
                href="/author/pedro-coelho"
              >
                Pedro Coelho
              </a>
            </span>
            <span>
              <svg
                aria-hidden="true"
                class="c6"
                fill="currentColor"
                focusable="false"
                height="16"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V10h16v11zm0-13H4V5h16v3z"
                />
              </svg>
              19 de abr. de 2020
            </span>
          </div>
          <h2
            class="c7 c8"
          >
            <a
              href="/futebol/al-sadd-anuncia-acordo-para-liberar-xavi-mas-barcelona-nega"
            >
              Al Sadd anuncia acordo para liberar Xavi, mas Barcelona nega
            </a>
          </h2>
          <p
            class="c9"
          >
            A novela continua: Barcelona vê em anúncio de acerto do Al Sadd como uma forma de pressão para o pagamento da multa.
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <a
            href="/futebol/com-dois-penaltis-napoli-vira-contra-o-legia-e-se-recupera-em-seu-grupo-da-liga-europa"
          >
            <img
              alt=""
              class="c4"
              src="https://res.cloudinary.com/rabbitz/image/upload/v1636129861/zielinski_napoli_9359de0d9d.jpg"
            />
          </a>
        </div>
        <div
          class=""
        >
          <div
            class="c5"
          >
            <span>
              <svg
                aria-hidden="true"
                class="c6"
                fill="currentColor"
                focusable="false"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
              <a
                href="/author/pedro-coelho"
              >
                Pedro Coelho
              </a>
            </span>
            <span>
              <svg
                aria-hidden="true"
                class="c6"
                fill="currentColor"
                focusable="false"
                height="16"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V10h16v11zm0-13H4V5h16v3z"
                />
              </svg>
              21 de abr. de 2020
            </span>
          </div>
          <h2
            class="c7 c8"
          >
            <a
              href="/futebol/com-dois-penaltis-napoli-vira-contra-o-legia-e-se-recupera-em-seu-grupo-da-liga-europa"
            >
              Com dois pênaltis, Napoli vira contra o Legia e se recupera em seu grupo da Liga Europa
            </a>
          </h2>
          <p
            class="c9"
          >
            Após levar apenas um ponto nas duas primeiras rodadas, o Napoli chegou a sete se colocou em situação mais confortável para a reta final.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`
